includes/Feed/FeedFileUploadHandler.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace SkyVerge\WooCommerce\Facebook\Feed;
+
+use SkyVerge\WooCommerce\PluginFramework\v5_10_0 as Framework;
+
+
+defined( 'ABSPATH' ) || exit;
+/**
+ * Class FeedFileUploadHandler
+ *
+ * @since 2.6.0
+ */
+class FeedFileUploadHandler {
```

Comment:
I have chosen this name to match FB nomenclature which is always referring to this operation as Upload.
The distinction made by FB is `push Upload` or `pull Upload` 🤷🏻 I can change it if this explanation is not convincing enough :)

Reply:
Yep I think we can make this clearer :) 

I do agree it's useful to make it clear in docs/comments what this corresponds to in the Facebook UI, but words like `upload` have a different meaning on each end.

Reply:
I think calling this class an `UploadHandler` suggests the class performs an upload e.g. via a web request, but it's actually doing the opposite by responding to a web request. `FeedApiEndpoint`?

Reply:
I don't really understand the objection to upload.  So if it is not making/handling the upload then who does that?
The procedure is called Feed Upload and this class handless our side of that procedure.

> I think calling this class an UploadHandler suggests the class performs an upload e.g. via a web request, but it's actually doing the opposite by responding to a web request.

The opposite to upload is download. I don't really think that we are doing anything like that in the class :)

> FeedApiEndpoint

Very vague connection to what the class is actually doing.

I am taking a step back. Probably it just means that I don't understand English well enough to grasp the problem with the name I have proposed 😉 @haszari @danielbitzer please help me by providing an alternative name, OK if you want `FeedApiEndpoint` so I can proceed.

Reply:
`FeedApiEndpoint` is a great fit IMO - let's go with that. This class implements an API endpoint to serve up the feed file. 

"Upload" means to load a file up from a satellite system (client/browser) to a remote server ("up", in the cloud). Upload implies a client-driven transfer. In this case, it's a transfer initiated by the remote end (Facebook pull). In any case, `upload` and `download` are problematic because they imply a client OR server context, which may depend on your point of view (since Facebook is a server, and the merchant woo site is also a server).

Reply:
🤷🏻

Reply:
renamed